test/y2partitioner/widgets/add_lvm_button_test.rb

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com
+
+require_relative "../test_helper"
+
+require "cwm/rspec"
+require "y2partitioner/widgets/add_lvm_button"
+
+describe Y2Partitioner::Widgets::AddLvmButton do
+  before do
+    devicegraph_stub("lvm-two-vgs.yml")
+  end
+
+  subject(:button) { described_class.new(table) }
+
+  let(:table) { double("table", selected_device: selected_device) }
+
+  let(:selected_device) { nil }
+
+  include_examples "CWM::PushButton"
+
+  describe "#handle" do
+    before do
+      allow(Y2Partitioner::Sequences::AddLvmLv).to receive(:new).and_return sequence
+    end
+
+    let(:sequence) { double("AddLvmLv", run: :result) }
+
+    let(:vg) { Y2Storage::LvmVg.find_by_vg_name(current_graph, "vg0") }
+
+    let(:lv) { vg.lvm_lvs.first }
+
+    let(:current_graph) { Y2Partitioner::DeviceGraphs.instance.current }
+
+    let(:event) { { "ID" => selected_option } }
+
+    context "when a vg is selected in the table" do
+      let(:selected_device) { vg }
+
+      context "and option for adding vg is selected" do
+        let(:selected_option) { :add_volume_group }
+
+        xit "opens the workflow for adding a new vg" do
```

Comment:
Maybe you already know: if you do not specify the `do...end` block, it will work as `xit`. The advantage of `xit` is that it could be easier to find it in your editor.

Reply:
I didn't know this trick either. As you mention, I will leave `xit` syntax to find it easier.